### ISSUE TITLE
`bazel run` before `bazel test` fails with `Could not chdir` on Windows

### ISSUE THREAD

**Issue by gregmagolan** [bazelbuild/bazel#10621]
### Description of the problem / feature request:

On Windows, if you run a test target with `bazel run` before you run that same test with `bazel test` (after a clean) then the test will fail with `Could not chdir` on Windows.

This was originally reported in #7387 and marked as resolved so it may have regressed as I'm seeing this in Bazel 2.0.0.

The culprit is that under Windows with runfiles not enabled by default, `bazel run` does not create the root workspace directory in the runfiles directory (`/path/to/test.runfiles/my_workspace_name`) while `bazel test` does. If you run `bazel test` first then `bazel run` will succeed for that test from then on as that directory is present under the test runfiles.

The error happens with both the default `--incompatible_windows_native_test_wrapper` and `--noincompatible_windows_native_test_wrapper` (although the error output is slightly different):

```
-----------------------------------------------------------------------------
ERROR(tools/test/windows/tw.cc:462) value: 2 (0x00000002), arg: c:\users\gmago\_bazel~1\wpufrwit\execroot\build_~1\bazel-out\x64_windows-fastbuild\bin\internal\linker\test\unit_tests.bat.runfiles\build_bazel_rules_nodejs: Could not chdir
```
in the first case, and

```
external/bazel_tools/tools/test/test-setup.sh: line 147: cd: /c/users/gmago/_bazel~1/wpufrwit/execroot/build_~1/bazel-out/x64_windows-fastbuild/bin/internal/linker/test/unit_tests.bat.runfiles/build_bazel_rules_nodejs: No such file or directory
Could not chdir /c/users/gmago/_bazel~1/wpufrwit/execroot/build_~1/bazel-out/x64_windows-fastbuild/bin/internal/linker/test/unit_tests.bat.runfiles/build_bazel_rules_nodejs
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

On Windows with Bazel 2.0.0, run any test target with `bazel run` on a clean clone or after a bazel clean before running that test target with `bazel test`.

### What operating system are you running Bazel on?

Windows 10 Pro

### What's the output of `bazel info release`?

```
C:\Users\gmago\google\rules_nodejs>bazel info release
INFO: Invocation ID: 1ecaee15-b295-461c-9656-0c5fe8418002
release 2.0.0
```

### Any other information, logs, or outputs that you want to share?

The work-around is simply to run `bazel test` first but it would be convenient if the root workspace directory was created in the runfiles folder under `bazel run` as well. 

//cc @devversion

**Comment by laszlocsomor** [comment]
Thanks for the report.
I can repro locally, with a trivial cc_test.

foo/BUILD:
```
cc_test(name = "x", srcs = ["x.cc"])
```

foo/x.cc:
```
int main() { return 0; }
```

```
bazel run //foo:x
(...)
Executing tests from //foo:x
-----------------------------------------------------------------------------
ERROR(tools/test/windows/tw.cc:462) value: 2 (0x00000002), arg: c:\_bazel\ansasmbz\execroot\__main__\bazel-out\x64_windows-fastbuild\bin\foo\x.exe.runfiles\__main__: Could not chdir
```

Your root cause analysis seems correct.

**Comment by laszlocsomor** [comment]
I have a fix for this.

The bug affects Linux too when using `--enable_runfiles=no`.

**Comment by gregmagolan** [comment]
Thanks for the quick fix @laszlocsomor !

**Comment by laszlocsomor** [comment]
You're welcome! ETA for the fix is Bazel 2.1.0.

**Comment by coryan** [comment]
@laszlocsomor Just curious, did #10646 make it to Bazel 2.1.0?  I was having this problem today on 2.1.0 and was wondering if it is a new issue or simply need to wait until 2.2.0 or something.

**Comment by laszlocsomor** [comment]
@coryan : It didn't; the baseline for 2.1.0 was 41ec5a28fb30a8d6c5c60194c4bb29528352cf78 from Jan 8 while #10646 was merged as 02ce5bd3d987b0a3474c1aa9e2450c70e71ab5d6 on Jan 28.

It will be in 2.2.0, whose baseline is 78055efad0917b848078bf8d97b3adfddf91128d from Feb 13.
